swatch-system-conduit/src/test/java/org/candlepin/subscriptions/conduit/InventoryControllerTest.java

```diff
@@ -946,4 +946,26 @@ void testSystemMemoryBytesOverMaxValueSetsToNull() throws ApiException {
     verify(inventoryService).scheduleHostUpdate(cfacts);
     verify(inventoryService, times(1)).flushHostUpdates();
   }
+
+  @Test
+  void testNumberOfCpusIsMappedToConduitFacts() {
+    int expectedNumberOfCpus = 5;
+
+    Consumer consumer = new Consumer();
+    consumer.getFacts().put("cpu.cpu(s)", "" + expectedNumberOfCpus);
+
+    ConduitFacts conduitFacts = controller.getFactsFromConsumer(consumer);
+    assertEquals(expectedNumberOfCpus, conduitFacts.getNumberOfCpus());
+  }
+
+  @Test
+  void testThreadsPerCoreIsMappedToConduitFacts() {
+    int expectedThreadsPerCore = 3;
+
+    Consumer consumer = new Consumer();
+    consumer.getFacts().put("cpu.thread(s)_per_core", "" + expectedThreadsPerCore);
+
+    ConduitFacts conduitFacts = controller.getFactsFromConsumer(consumer);
+    assertEquals(expectedThreadsPerCore, conduitFacts.getThreadsPerCore());
+  }
```

Comment:
I'd like to see tests covering the case where the fact is not set if it is not present.

Reply:
Added